internal/provider/resource_point_to_point.go

```diff
@@ -142,49 +141,81 @@ func resourcePointToPointCreate(ctx context.Context, d *schema.ResourceData, m i
 		return diag.FromErr(err)
 	}
 	if resp != nil {
+		_ = d.Set("ptp_circuit_id", resp.PtpCircuitID)
 		d.SetId(resp.PtpUUID)
 	}
 	return diags
 }
 
 func resourcePointToPointRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	c := m.(*packetfabric.PFClient)
-	c.Ctx = ctx
-	var diags diag.Diagnostics
-	if cID, ok := d.GetOk("ptp_circuit_id"); ok {
-		if _, err := c.GetPointToPointInfo(cID.(string)); err != nil {
-			return diag.FromErr(err)
-		}
-	}
-	return diags
+	return diag.Diagnostics{}
 }
 
 func resourcePointToPointUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*packetfabric.PFClient)
 	c.Ctx = ctx
 	var diags diag.Diagnostics
+
+	err := checkUpdatableFieldsPTP(ctx, d, m)
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Error updating resource",
+			Detail:   err.Error(),
+		})
+		return diags
+	}
+
 	if desc, ok := d.GetOk("description"); ok {
 		if _, err := c.UpdatePointToPoint(d.Id(), desc.(string)); err != nil {
 			return diag.FromErr(err)
 		}
 	}
+
+	if subTerm, ok := d.GetOk("subscription_term"); ok {
+		return diag.Errorf("please provide a subscription term")
+	} else {
+		billing := packetfabric.BillingUpgrade{
+			SubscriptionTerm: subTerm.(int),
+		}
+		cID := d.Get("ptp_circuit_id").(string)
+		if _, err := c.ModifyBilling(cID, billing); err != nil {
+			return diag.FromErr(err)
+		}
+		_ = d.Set("subscription_term", subTerm.(int))
+	}
 	return diags
 }
 
+func checkUpdatableFieldsPTP(ctx context.Context, d *schema.ResourceData, m interface{}) error {
+	c := m.(*packetfabric.PFClient)
+	c.Ctx = ctx
+
+	if d.HasChange("speed") ||
+		d.HasChange("media") ||
+		d.HasChange("pop") ||
+		d.HasChange("zone") ||
+		d.HasChange("autoneg") ||
+		d.HasChange("published_quote_line_uuid") {
+		return errors.New("only the description or subscription term field can be updated")
```

Comment:
Shouldn't we just set `ForceNew` to `true` for attributes here? It is not possible to edit, but it is possible to create a new resource with these values.

Reply:
I think this is a great suggestion @medzin I tried with https://github.com/PacketFabric/terraform-provider-packetfabric/pull/323 and it works pretty well. I will update this PR too.

Reply:
@medzin I implemented the `ForceNew()`, really good suggestion